scripts/manage-isos.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+set -u
+
+usage()
+{
+    echo "----------------------------------------------------------------------------------------"
+    echo "- Usage: $0 [version XCP-ng] [Repository] [Server PXE] [Path on the server PXE] [Server SR ISO] [Path on the server of the SR ISO] [Path on the pXE for th installer]"
+    echo "- All options are mandatory."
+    echo "- Repository: updates, ci, testing..."
+    echo "- Version of XCP-ng: 8.2 or 8.3"
+    echo "- Example: bash manage-isos.sh 8.3 myuser@my_pxe /my/path/pxe/ iso_user@srv-sr-iso /my/path/on/sriso/ /my/path/pxe/installers/"
+    echo "----------------------------------------------------------------------------------------"
+}
+
+VERSION=$1
+REPO=$2
+SERVERPXE=$3
+PATHPXE=$4
+SRISO=$5
+PATHSRISO=$6
+PATHPXEINST=$7
+
+if [ $VERSION != "8.3" ] && [ $VERSION != "8.2" ]; then
+    echo "Unsupported version. Please choose between 8.2 or 8.3."
+    exit
+fi
+
+if [ -z ${PATHPXE} ]; then
+    echo "PATHPXE is empty. Please provide the path on the pxe server."
+    exit
+fi
+
+if [ -z ${PATHSRISO} ]; then
+    echo "PATHSRISO is empty. Please provide the path on the SR ISO."
+    exit
+fi
+
+if [ -z ${PATHPXEINST} ]; then
+    echo "PATHPXEINST is empty. Please provide the path on the PXE for the installers."
+    exit
+fi
+
+#get files names
+FILE_NAME_ISO_NI=$(ls *.iso | grep netinstall)
+FILE_NAME_ISO=$(ls *.iso | grep -v netinstall)
+
+#scp to the pxe server
+echo "We're doing the scp to the pxe server for the two isos."
+scp *.iso ${SERVERPXE}:${PATHPXE}
+echo "Creating or changing the link latest on the pxe."
+ssh ${SERVERPXE} 'ln -sf '${PATHPXE}${FILE_NAME_ISO}' '${PATHPXE}'xcp-ng-'${VERSION}'-testing-latest'
+echo "Cleaning iso older than 10 days on the pxe."
+ssh ${SERVERPXE} 'find '${PATHPXE}' -mtime +9 -name *nightly*.iso -delete'
+
+if [ $REPO == "ci" ] || [ $REPO == "updates" ]; then
+    echo "Repo: ${REPO}: copy the iso content (${FILE_NAME_ISO}) into a directory of the pxe server."
+    NAME_LOCAL_DIRECTORY="/tmp/"${FILE_NAME_ISO%.*}
+    NAME_DISTANT_DIRECTORY="${PATHPXEINST}${VERSION}-${REPO}/"
+    mkdir ${NAME_LOCAL_DIRECTORY}
+    7z x -o${NAME_LOCAL_DIRECTORY}/ ${FILE_NAME_ISO}
+    if [ ! -z ${NAME_DISTANT_DIRECTORY} ]; then
+        rsync -av --delete "${NAME_LOCAL_DIRECTORY}/" "${SERVERPXE}:${NAME_DISTANT_DIRECTORY}"
+        ssh ${SERVERPXE} "chmod -R 755 ${NAME_DISTANT_DIRECTORY}"
+    fi
+    if [ -d ${NAME_LOCAL_DIRECTORY} ]; then
+        rm -Rf ${NAME_LOCAL_DIRECTORY}
+    fi
+fi
+
+#Manage the monthly one on the first of each month
+if [[ $(date +%d) == 01 ]]; then
+    echo "We're the first of the month, so we create a monthly iso to keep 3 months."
+    NEW_NAME_ISO=${FILE_NAME_ISO/"nightly"/"monthly"}
+    scp ${FILE_NAME_ISO} ${SERVERPXE}:${PATHPXE}${NEW_NAME_ISO}
```

Comment:
Since everything is on a single drive, a hardlink would save a bit of disk space and transfer time (bonus: `ls -l` in nightly directory would show that a nightly is also a monthly)